src/Listener/ApiPaginationListener.php

```diff
@@ -70,4 +78,60 @@ public function beforeRender(Event $event)
         $controller->set('pagination', $paginationResponse);
         $this->_action()->config('serialize.pagination', 'pagination');
     }
+
+    /**
+     * Generates pagination viewVars with JSON API compatible hyperlinks.
+     *
+     * @param array $pagination CakePHP pagination result
+     * @return array
+     */
+    protected function _getJsonApiPaginationResponse(array $pagination)
```

Comment:
I had that once but deliberately chose to integrate into ApiPaginationListener as this keeps things consistent when setting up. E.g. here's how I am testrunning it. All I need to do to switch between the listeners is change `JsonApi` to `Api`.

```php
        $this->_controller->loadComponent('Crud.Crud', [
            'actions' => [
                'Crud.Index',
                'Crud.View',
                'Crud.Add',
                'Crud.Edit',
                'Crud.Delete'
            ],
            'listeners' => [
                'Crud.JsonApi',
                'Crud.ApiPagination',
                'Crud.ApiQueryLog'
            ]
        ]);
```